presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -2382,6 +2382,26 @@ public void dropMaterializedView(ConnectorSession session, SchemaTableName viewN
         }
     }
 
+    @Override
+    public List<String> getValidRefreshMaterializedViewFilterColumns(ConnectorSession session, SchemaTableName viewName)
+    {
+        requireNonNull(viewName, "viewName is null");
+
+        Optional<Table> viewTable = metastore.getTable(viewName.getSchemaName(), viewName.getTableName());
+
+        if (!viewTable.isPresent() || !MetastoreUtil.isPrestoMaterializedView(viewTable.get())) {
+            throw new MaterializedViewNotFoundException(viewName);
+        }
+
+        List<String> partitionedBy = viewTable.get().getPartitionColumns().stream()
+                .map(Column::getName)
+                .collect(toImmutableList());
```

Comment:
No need to figure out whether it's a partition column or not. Use the full mapping

Reply:
I think, for REFRESH, we only allow mv partitions in the where clause instead of random columns.